crates/router/src/connector/zen/transformers.rs

```diff
@@ -345,26 +345,76 @@ fn get_item_object(
 }
 
 fn get_browser_details(
-    browser_info: &BrowserInformation,
-) -> Result<ZenBrowserDetails, error_stack::Report<errors::ConnectorError>> {
-    let window_size = match (browser_info.screen_height, browser_info.screen_width) {
+    browser_info: &types::BrowserInformation,
+) -> CustomResult<ZenBrowserDetails, errors::ConnectorError> {
+    let screen_height = browser_info
+        .screen_height
+        .get_required_value("screen_height")
+        .change_context(errors::ConnectorError::MissingRequiredField {
+            field_name: "screen_height",
+        })?;
+
+    let screen_width = browser_info
+        .screen_width
+        .get_required_value("screen_width")
+        .change_context(errors::ConnectorError::MissingRequiredField {
+            field_name: "screen_width",
+        })?;
+
+    let window_size = match (screen_height, screen_width) {
         (250, 400) => "01",
         (390, 400) => "02",
         (500, 600) => "03",
         (600, 400) => "04",
         _ => "05",
     }
     .to_string();
+
     Ok(ZenBrowserDetails {
-        color_depth: browser_info.color_depth.to_string(),
-        java_enabled: browser_info.java_enabled,
-        lang: browser_info.language.clone(),
-        screen_height: browser_info.screen_height.to_string(),
-        screen_width: browser_info.screen_width.to_string(),
-        timezone: browser_info.time_zone.to_string(),
-        accept_header: browser_info.accept_header.clone(),
+        color_depth: browser_info
+            .color_depth
+            .get_required_value("color_depth")
+            .change_context(errors::ConnectorError::MissingRequiredField {
+                field_name: "color_depth",
+            })?
```

Comment:
Are all these fields really mandatory? can you check the api documentation and confirm this?